File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> hash_map;
    
    for (string name : participant) {
        hash_map[name]++;
    }
    
    for (string name : completion) {
        hash_map[name]--;
    }
    
    for (auto pair : hash_map) {
        if (pair.second > 0) {
            return pair.first;
        }
    }
    return "";
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> map = new HashMap<>();
        String answer = "";

        for (int i = 0; i < participant.length; i++) {
            if (map.containsKey(participant[i])) {
                map.put(participant[i], map.get(participant[i]) - 1);
            } else {
            map.put(participant[i], 0);
            }
        }
        
        for (String name : completion) {
            map.put(name, map.get(name) + 1);
        }
        
        for (String name : participant) {
            if (map.get(name) == 0) {
                answer += name;
                break;
            }
        } 
        return answer;
    }
}